src/pudl/output/eia923.py

```diff
@@ -230,11 +255,12 @@ def denorm_boiler_fuel_eia923(
         boiler_fuel_eia923["fuel_consumed_units"]
         * boiler_fuel_eia923["fuel_mmbtu_per_unit"]
     )
-    return denorm_by_boil(
+    dd = denorm_by_boil(
         boiler_fuel_eia923,
-        pu=denorm_plants_utilities_eia,
+        pu=denorm_plants_utilities_eia.drop(columns=["data_maturity"]),
         bga=boiler_generator_assn_eia860,
     )
```

Comment:
this could be a bad suggestion bc i don't know all the places `denorm_by_X` are used but is it possible to do these drop `data_maturity`'s in there instead of having them sprinkled throughout?

Reply:
I think this is a good idea. I restructured the code to reflect this.